lib/sufia/arkivo/actor.rb

```diff
@@ -0,0 +1,96 @@
+require 'base64'
+require 'tempfile'
+
+# Ugly monkeypatch to make Tempfiles quack like UploadedFiles
+Tempfile.class_eval do
+  attr_accessor :original_filename, :content_type
+end
+
+module Sufia
+  module Arkivo
+    class Actor
+      attr_reader :user, :item
+
+      def initialize(user, item)
+        @user = user
+        @item = item
+      end
+
+      def create_file_from_item
+        batch = Batch.create
+        file_actor = Sufia::GenericFile::Actor.new(::GenericFile.new, user)
+        file_actor.create_metadata(batch.id)
+        store_checksum(file_actor.generic_file)
+        file_actor.create_content(file, item['file']['filename'], file_path, item['file']['contentType'])
+        ::Resque.inline = true
+        Sufia.queue.push(BatchUpdateJob.new(user.user_key, batch.id, item['metadata']['title'], attributes, default_visibility))
```

Comment:
Ah, do we have an example of that syntax somewhere?  I briefly looked for an example of it, couldn't find it, and used inline and moved on.  Happy to rework this for clarity.

(The reason I run the job synchronously here is that the request that calls this method is sent from the Arkivo server, not a real human user, so I thought a slight delay would be acceptable.)